Elcano_C2_LowLevel/Elcano_C2_LowLevel.ino

```diff
@@ -236,119 +263,156 @@ void setup()
+  // an inconsistent state.)
+  for (int i = 0; i < RC_NUM_SIGNALS; i++)
     local_results[i] = RC_elapsed[i];
-  //Print7(false, local_results);
-  //Serial.println("Processing RC data...");
+
   byte automate = processRC(local_results);
-  if (automate == 0x01) //remember to tell Pat to fix this, was = instead of ==
+  // @ToDo: Verify that this should be conditional. May be moot if it is
+  // replaced in the conversion to the new Elcano Serial protocol.
+  if (automate == 0x01)
   {
-    //Serial.println("Processing high-level data");
     processHighLevel(&Results);
   }
-  //    Print7( true, local_results);
-    
-  //Serial.println(nextTime); //Old sanity check line; testing new pin output sanity check.
-  //Serial.println("Clearing Results");
-//  Results.Clear();
-//  Results.kind = MSG_SENSOR;
-//  Results.angle_deg = TurnAngle_degx10() / 10;
-  //show_speed (&Results);
-
-  // Report how long the loop took.
-  
-  digitalWrite(27, HIGH);
-  if(nextTime < startTime || nextTime > startTime + LOOP_TIME_US) {
-    digitalWrite(33, HIGH);
-  }
-  //Serial.println(nextTime); //Old sanity check line; testing new pin output sanity check.
-  //Serial.println("Clearing Results");
+
+  // @ToDo: What is this doing?
   //Results.clear();
   Results.kind = MSG_SENSOR;
   Results.angle_deg = TurnAngle_degx10() / 10;
-  //show_speed (&Results);
-
-  //LogData(local_results, &Results);  // data for spreadsheet
-
-  calibrationTime_ms += LOOP_TIME_US;
-  straightTime_ms = (steer_control == STRAIGHT_TURN_OUT) ? straightTime_ms + LOOP_TIME_US : 0;
-  stoppedTime_ms = (throttle_control == MIN_ACC_OUT) ? stoppedTime_ms + LOOP_TIME_US : 0;
-  if (calibrationTime_ms > 40000 && straightTime_ms > 3000 && stoppedTime_ms > 3000)
-  {
-    //       int oldBrake = brake_control;
-    //       brake(MAX_BRAKE_OUT);  // put on brakes
-    CalibrateTurnAngle(16, 10);  // WARNING: No response to controls while calibrating
-    //       brake(oldBrake);       // restore brake state
-    calibrationTime_ms = 0;
+  // @ToDo: Is this working and should it be uncommented?
+  // show_speed (&Results);
+
+  // @ToDo: Remove all test or calibration code. If needed, put it in a separate file.
+  //calibrationTime_ms += LOOP_TIME_US;
+  //straightTime_ms = (steer_control == STRAIGHT_TURN_OUT) ? straightTime_ms + LOOP_TIME_US : 0;
+  //stoppedTime_ms = (throttle_control == MIN_ACC_OUT) ? stoppedTime_ms + LOOP_TIME_US : 0;
+  //if (calibrationTime_ms > 40000 && straightTime_ms > 3000 && stoppedTime_ms > 3000)
+  //{
+  //  //       int oldBrake = brake_control;
+  //  //       brake(MAX_BRAKE_OUT);  // put on brakes
+  //  CalibrateTurnAngle(16, 10);  // WARNING: No response to controls while calibrating
+  //  //       brake(oldBrake);       // restore brake state
+  //  calibrationTime_ms = 0;
+  //}
+
+  // @ToDo: What information do we need to send to C6? Is that communication already
+  // hiding in here somewhere, or does it need to be added?
+
```

Comment:
C6 needs speed and turn angle. In the original ElcanoSerial, it is sent by clearing results, setting results.kind = MSG_SENSOR, filling in correct data for speed and/or turn angle, then doing a write.